src/PointwiseFunctions/InitialDataUtilities/InitialData.hpp

```diff
@@ -0,0 +1,33 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <pup.h>
+#include <pup_stl.h>
+
+#include "Parallel/CharmPupable.hpp"
+
+namespace evolution {
+/*!
+ * \brief Namespace for things related to initial data used for evolution
+ * systems.
+ */
+namespace initial_data {
+/*!
+ * \brief The abstract base class for initial data of evolution systems.
```

Comment:
Maybe add a comment like `All analytic solutions and analytic data must virtually inherit from this class`? I think we discussed that they probably all need virtual inheritance. I expect this will be non-intuitive and so should be documented.

Reply:
Are you sure this is needed? Should only be needed if we have solution/data classes that inherit from multiple other solution/data classes, I can't think of one.

Reply:
Well the recommendation is that all analytic data and solutions inherit from this. There are analytic data that inherit from analytic solutions, which should also explicitly inherit from this base class since the inheritance of the analytic solution is typically private. I believe that leaves the base classes of the analytic solution also private. Public inheritance of the solution is definitely incorrect and will cause problems.